pkg/restore/restore.go

```diff
@@ -1507,6 +1507,22 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		}
 	}
 
+	// Check if cluster has GVK required to restore resource, otherwise return err.
+	// Placed here because the object apiVersion might get modified by a RestorePlugin
+	// So we check for the GVK after the RestorePlugin has run.
+	_, _, err = ctx.discoveryHelper.KindFor(schema.GroupVersionKind{
+		Group:   groupResource.Group,
+		Version: obj.GetAPIVersion(),
+		Kind:    groupResource.Resource,
+	})
+	clusterHasKind := err == nil
+	if !clusterHasKind  {
+		ctx.log.Errorf("Cannot restore %s because GVK doesn't exist in the cluster", groupResource)
+		// Adding to errs should cause restore to partially fail
+		errs.Add(namespace, fmt.Errorf("cannot restore %s because gvk doesn't exist in the cluster", groupResource))
```

Comment:
This should cause restore to partially fail, but the error message maybe different than before. Alternatively, [another commits](https://github.com/vmware-tanzu/velero/commit/6c4bb32b8bc1bcda9ade8640593c77a248f1d782) attempts to preserve prior error message for when informer cache is disabled. Let me know which is preferred.